### PR TITLE
RavenDB: Add version 4.2.4

### LINKED ISSUE #511: [Request] add RavenDB
https://ravendb.net/

RavenDB, NoSQL database built on .NET Core.
It would depend on dotnet-sdk, I suppose.
Downloads from here
https://ravendb.net/download
include the option to download as a .zip file

### PR DESCRIPTION
**Pull request by issaclin32**
Close #511

**RavenDB** ([homepage](https://ravendb.net/)) is a NoSQL database built on .NET Core.

Please let me know if there are any problems. Thanks.

### CODE REVIEW

File: bucket/ravendb.json [hunk 1]
```diff
@@ -0,0 +1,53 @@
+{
+    "homepage": "https://ravendb.net/",
```
Comment (Ash258) [inline bucket/ravendb.json]:
```suggestion
    "homepage": "https://ravendb.net",
```

File: bucket/ravendb.json [hunk 2]
```diff
@@ -0,0 +1,53 @@
...
+    "description": "NoSQL database built on .NET Core.",
+    "version": "4.2.4",
+    "license": {
+        "identifier": "Freeware",
+        "url": "https://ravendb.net/terms"
```
Comment (linsui) [inline bucket/ravendb.json]:
````suggestion
        "url": "https://ravendb.net/terms/commercial"
```

File: bucket/ravendb.json [hunk 3]
```diff
@@ -0,0 +1,55 @@
...
+        "32bit": {
+            "url": "https://daily-builds.s3.amazonaws.com/RavenDB-4.2.4-windows-x86.zip",
+            "hash": "e93516db0f8f37772de490068a655fc7b3794b6061b5c8abffaafa9dde4fd197"
+        },
+        "64bit": {
+            "url": "https://daily-builds.s3.amazonaws.com/RavenDB-4.2.4-windows-x64.zip",
+            "hash": "8e65f887a677e78a92e57faf71363c17a1b2bf54b0c490f5b2618b597117dd99"
+        }
+    },
+    "pre_install": "if(!(Test-Path $persist_dir\\Server\\settings.json)) {Set-Content $dir\\Server\\settings.json -Value $null}",
```
Comment (linsui) [inline bucket/ravendb.json]:
```suggestion
    "pre_install": "if(!(Test-Path \"$persist_dir\\Server\\settings.json\")) {Set-Content \"$dir\\Server\\settings.json\" -Value $null}",
```

File: bucket/ravendb.json [hunk 4]
```diff
@@ -7,7 +7,7 @@
         "url": "https://ravendb.net/terms/commercial"
     },
     "suggest": {
-        ".NET SDK": "dotnet-sdk"
+        "Visual C++ Redistributable 2015": "extras/vcredist2015"
```
Comment (Ash258) [inline bucket/ravendb.json]: Why did you removed dotnet?

File: bucket/ravendb.json [hunk 5]
```diff
@@ -19,6 +19,7 @@
             "hash": "8e65f887a677e78a92e57faf71363c17a1b2bf54b0c490f5b2618b597117dd99"
         }
     },
+    "pre_install": "if(!(Test-Path $persist_dir\\settings.json)) {Set-Content $dir\\settings.json -Value $null}",
```
Comment (Ash258) [inline bucket/ravendb.json]: - Unqoted path
- Do not use $null in set-content
    - Use new-item or set it value
- Always use encoding in set-content

File: bucket/ravendb.json [hunk 6]
```diff
@@ -36,6 +37,7 @@
             "raven-uninstall-service"
         ]
     ],
+    "persist": "settings.json",
```
Comment (Ash258) [inline bucket/ravendb.json]: There is said it is inside `Server` folder

> The settings.json file which can be found in the same directory as the server executable 

> This file is located in the Server directory

File: bucket/ravendb.json [hunk 7]
```diff
@@ -0,0 +1,55 @@
...
+        "32bit": {
+            "url": "https://daily-builds.s3.amazonaws.com/RavenDB-4.2.4-windows-x86.zip",
+            "hash": "e93516db0f8f37772de490068a655fc7b3794b6061b5c8abffaafa9dde4fd197"
+        },
+        "64bit": {
+            "url": "https://daily-builds.s3.amazonaws.com/RavenDB-4.2.4-windows-x64.zip",
+            "hash": "8e65f887a677e78a92e57faf71363c17a1b2bf54b0c490f5b2618b597117dd99"
+        }
+    },
+    "pre_install": "if(!(Test-Path \"$persist_dir\\Server\\settings.json\")) {Set-Content \"$dir\\Server\\settings.json\" -Value $null}",
```
Comment (Ash258) [inline bucket/ravendb.json]: - Do not use $null in set-content
    - Use new-item instead
- Always use encoding while using set-content

**Pull request merged**